core/shared/src/main/scala/zio/prelude/recursive/Coalgebra.scala

```diff
@@ -0,0 +1,27 @@
+package zio.prelude.recursive
+
+import zio.prelude._
+
+sealed trait Coalgebra[Case[+_], A] extends (A => Case[A]) { self =>
+
+  def zipWith[B](that: Coalgebra[Case, B])(implicit both: AssociativeBoth[Case]): Coalgebra[Case, (A, B)] =
```

Comment:
Else where we use `zipWith` for zipping other data structure _with some function_. Could we name this `contrazip`, `cozip` or something like that? :thinking: 

I'm not really sure, just thinking out loud.

Reply:
Oh I think we should just rename it `zip`.

Reply:
We won't have an object which is an instance of both algebra and co-algebra, so there's no risk of a clash, right?

Reply:
Right.